detectable/src/main/java/com/synopsys/integration/detectable/detectables/maven/cli/MavenCliExtractorOptions.java

```diff
@@ -28,6 +43,44 @@ public Optional<String> getMavenBuildCommand() {
         return Optional.ofNullable(mavenBuildCommand);
     }
 
+    public List<String> buildCliArguments(CommandParser commandParser) {
+        List<String> arguments = new ArrayList<>();
+        List<String> passedArgs = commandParser.parseCommandString(getMavenBuildCommand().orElse(""));
+
+        boolean omitArg = false;
+        boolean foundThreadArguments = false;
+        for (String arg : passedArgs) {
+            if (omitArg || arg.equals("dependency:tree")) {
+                omitArg = false;
+                continue;
+            }
+
+            boolean isThreadSpecifier = false;
+
+            for (String prefix : THREAD_SPECIFYING_ARGUMENT_PREFIXES) {
+                if (arg.startsWith(prefix)) {
+                    foundThreadArguments = true;
+                    isThreadSpecifier = true;
+                    omitArg = arg.length() == prefix.length(); // value can either be specified as part of the same argument or next
+                    break;
+                }
+            }
+            if (isThreadSpecifier)
+                continue;
+            arguments.add(arg);
+        }
+
+        if (foundThreadArguments)
+            logger.info("Thread-specifying Maven arguments provided by the user will be omitted to ensure that the dependency tree is generated accurately using a single thread");
```

Comment:
On behalf of potentially confused users, when thread related commands are ignored, thank you for logging this!
Another way to say this could be the following, though what you have is also good. (Just throw a period on the end. ; )  )
"To ensure that the dependency tree is generated with precision, user-supplied Maven thread-specifying arguments are omitted."